apps/evm/src/App/ChainUpgradeHandler/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,37 @@
+import { renderComponent } from 'testUtils/render';
+
+import { ChainUpgradeHandler } from 'App/ChainUpgradeHandler';
+import type { MockInstance } from 'vitest';
+
+const runtimeTimestamp = new Date().getTime();
+const upgradeTimestamps = [new Date(runtimeTimestamp + 2000)];
+const fakeNowMs = runtimeTimestamp + 12000;
+
+describe('ChainUpgradeHandler', () => {
+  let reloadSpy: MockInstance;
+
+  beforeEach(() => {
+    // 1. Redefine the 'reload' property on window.location to be configurable.
+    // This is necessary because it is non-configurable by default in jsdom.
+    Object.defineProperty(window, 'location', {
+      configurable: true,
+      value: () => {}, // Provide a dummy function initially
+    });
+
+    Object.defineProperty(window.location, 'reload', {
+      configurable: true,
+      value: () => {}, // Provide a dummy function initially
+    });
```

Comment:
I wonder if this test suite should reset the window.location.reload property after running (so in an `afterEach` function call)?